apps/schematic/api/schematic_api/controllers/schema_controller_impl.py

```diff
@@ -1,42 +1,91 @@
+from schematic.schemas.generator import SchemaGenerator, SchemaExplorer  # type: ignore
 
 from schematic_api.models.basic_error import BasicError
 from schematic_api.models.attributes_page import AttributesPage
 from schematic_api.models.attribute import Attribute
+from schematic_api.models.validation_rules_page import ValidationRulesPage
+from schematic_api.models.validation_rule import ValidationRule
+from schematic_api.models.nodes_page import NodesPage
+from schematic_api.models.node import Node
 from schematic_api.controllers.utils import handle_exceptions
 
 
-def get_component_attributes(
-    component_label: str,
+def get_node_label_from_schematic(
+    node_display: str, schema_url: str, use_strict_camel_case: bool
+) -> str:
+    """Gets the label of the node
+
+    Args:
+        node_display(str): The display name of the node
+        schema_url (str): The URL of the schema in jsonld form
+        use_strict_camel_case (bool): whether or not to use strict camel case when doing the
+          conversion
+
+    Returns:
+        str: The node label
+    """
+    schema_explorer = SchemaExplorer()
+    schema_explorer.load_schema(schema_url)
+    return schema_explorer.get_property_label_from_display_name(
+        node_display, use_strict_camel_case
+    )
+
+
+@handle_exceptions
+def get_node_label(
+    node_display: str, schema_url: str, use_strict_camel_case: bool
+) -> tuple[Union[str, BasicError], int]:
+    """Gets the label of the node
+
+    Args:
+        node_display(str): The display name of the node
+        schema_url (str): The URL of the schema in jsonld form
+        use_strict_camel_case (bool): whether or not to use strict camel case when doing the
+          conversion
+
+    Returns:
+        tuple[Union[str, BasicError], int]: A tuple
+          The first item is either the label or an error object
+          The second item is the response status
+    """
+    result: Union[str, BasicError] = get_node_label_from_schematic(
+        node_display, schema_url, use_strict_camel_case
+    )
+    status = 200
+    return result, status
+
+
+def get_node_attributes(
+    node_label: str,
     schema_url: str,
 ) -> list[str]:
-    """Gets the attributes associated with the component
+    """Gets the attributes associated with the node
 
     Args:
         schema_url (str): The URL of the schema in jsonld form
-        component_label (str): The label of the component
+        node_label (str): The label of the node
 
     Returns:
-        list[str]: A list of attributes of the component
+        list[str]: A list of attributes of the node
     """
     schema_explorer = SchemaExplorer()
     schema_explorer.load_schema(schema_url)
-    return schema_explorer.find_class_specific_properties(component_label)
+    return schema_explorer.find_class_specific_properties(node_label)
 
 
 @handle_exceptions
-def list_component_attributes(
-    component_label: str,
+def list_node_attributes(
```

Comment:
Same comment as above this is getting the properties associate with the node or attribute.